Add FPort support

fiam opened the pull request:
~Depends on #2902~ #2902 is now merged

FPort code is mostly imported from Betaflight. To make future RX
updates easier, a few more changes have been done to make the
RX/Telemetry code in BF and INAV a bit closer.

- Imported RSSI handling code from BF with some changes
- frsky_vfas_cell_voltage setting renamed to report_cell_voltage
- Added telemetry_halfduplex setting
- sbus_inversion renamed to serialrx_inverted, should apply to all
protocols eventually
- New MSP cmds MSP_TX_INFO/MSP_SET_TX_INFO now allow setting the RSSI
via MSP, thus RX_MSP doesn't need to use an extra channel anymore.

Fixes #2604

fiam committed:
Add FPort support

FPort code is mostly imported from Betaflight. To make future RX
updates easier, a few more changes have been done to make the
RX/Telemetry code in BF and INAV a bit closer.

- Imported RSSI handling code from BF with some changes
- frsky_vfas_cell_voltage setting renamed to report_cell_voltage
- Added telemetry_halfduplex setting
- sbus_inversion renamed to serialrx_inverted, should apply to all
protocols eventually
- New MSP cmds MSP_TX_INFO/MSP_SET_TX_INFO now allow setting the RSSI
via MSP, thus RX_MSP doesn't need to use an extra channel anymore.

digitalentity commented on src/main/rx/rx.c:
```diff
@@ -435,6 +512,50 @@ void parseRcChannels(const char *input)
+#define RSSI_SAMPLE_COUNT 16
+#define RSSI_MAX_VALUE 1023
+
+void setRssiUnfiltered(uint16_t rssiValue, rssiSource_e source)
```
Since we are introducing this function it would be good to make other RSSI code to use it as well (channel/analog)

digitalentity reviewed (commented):
This seems a direct grafting from Betaflight. I don't like the interrupt-driven implementation.

fiam committed:
Use new RSSI system to handle all RSSI sources

fiam committed:
Make MSP_SET_TX_INFO not crash on malformed MSP messages

Since it will be used while the machine is armed and over telemetry,
we need to make sure a malformed request doesn't crash the FC.

Also, rename setRSSIMsp() to setRSSIFromMSP()

fiam committed:
Replace setRSSIUnfiltered() and setRSSIFiltered() with a single function

Use a boolean to indicate wheter the value should be filtered. API
feels less confusing this way.
